### PR TITLE
#163519094 Setup react router 

### PR DESCRIPTION
**Pull request by NkFab**
### What does this PR do?
Sets up react router dom v4

### Description of Task to be completed?
Install package react-react-dom
Add Two files in a created view directory index.js and login.js
Add two directories Button, NavBar in components directory

### How should this be manually tested?

-Clone the repo
-Pull this branch
-run `npm install` or `yarn`
-run `npm run start` or `yarn start`

Any background context you want to provide?
N/A

What are the relevant pivotal tracker stories?
#163519094

Screenshots
N/A

Questions:
N/A

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    return (
+      <div>
+        <h1 className="app-name">Author's Haven</h1>
+        <h2 className="team-name">LIT Kigali</h2>
+        <Routes />
+      </div>
+    );
+  }
+}
+// const App = () => (
```
Comment (kimotho-njoki) [inline src/components/App.jsx]: why did we push our work with commented lines?

Comment (kimotho-njoki) [reply src/components/App.jsx]: App always has to be a stateful component...

File: src/views/Login.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React, { Component } from "react";
```
Comment (kimotho-njoki) [inline src/views/Login.js]: `Component` was imported but never used

File: src/views/Login.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (kimotho-njoki) [inline src/views/Login.js]:
```suggestion
const LoginComponent = () => (
  <h1 style={{ background: 'blue', color: 'white' }}>Login Component</h1>
);
```

Refactor to this. When using a functional component that has a single return value you do not need the `{ }` or the `return statement`. The `( )` automatically resolve to a return.

File: src/views/index.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React, { Component } from "react";
+
+const HomeComponent = () => {
+  return <h1 style={{ background: "gray", color: "white" }}>Home Component</h1>;
+};
```
Comment (kimotho-njoki) [inline src/views/index.js]: Same case here, no need for `{ } and return`

File: src/components/App.jsx (hunk 1 above)
Comment (NkFab) [reply src/components/App.jsx]: Ok I am to fix that.

File: src/views/Login.js (hunk 2 above)
Comment (NkFab) [reply src/views/Login.js]: Ok let me remove it

File: src/components/App.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  constructor() {
+    super();
+
+    this.state = {
+      appName: '',
+      teamName: ''
+    };
+  }
+
+  componentDidMount() {
```
Comment (oesukam) [inline src/components/App.jsx]: @NkFab Why do you need componentDidMount? Can't you just provide the default value when initializing the state?

**Review by oesukam — CHANGES_REQUESTED** [review]

**Pull request merged**